ui/pages/permissions-connect/snap-install/snap-install.js

```diff
@@ -0,0 +1,112 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { PageContainerFooter } from '../../../components/ui/page-container';
+import PermissionsConnectPermissionList from '../../../components/app/permissions-connect-permission-list';
+import PermissionsConnectFooter from '../../../components/app/permissions-connect-footer';
+import PermissionConnectHeader from '../../../components/app/permissions-connect-header';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { hideModal, showSnapInstallWarning } from '../../../store/actions';
+
+export default function SnapInstall({
+  request,
+  approveSnapInstall,
+  rejectSnapInstall,
+  targetSubjectMetadata,
+}) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  const onCancel = useCallback(() => rejectSnapInstall(request.metadata.id), [
+    request,
+    rejectSnapInstall,
+  ]);
+  const onSubmit = useCallback(() => approveSnapInstall(request), [
+    request,
+    approveSnapInstall,
+  ]);
+  const showWarning = useCallback(
+    () =>
+      dispatch(
+        showSnapInstallWarning(targetSubjectMetadata.name, () => {
+          dispatch(hideModal());
+          approveSnapInstall(request);
+        }),
+      ),
+    [request, targetSubjectMetadata.name, approveSnapInstall, dispatch],
+  );
+
+  const npmId = useMemo(() => {
+    if (!targetSubjectMetadata.origin.startsWith('npm:')) {
+      return undefined;
+    }
+    return targetSubjectMetadata.origin.substring(4);
+  }, [targetSubjectMetadata]);
+  const shouldShowWarning = useMemo(
+    () =>
+      Boolean(
+        Object.keys(request.permissions).find((v) =>
+          v.startsWith('snap_getBip44Entropy_'),
+        ),
+      ),
+    [request.permissions],
+  );
+
+  return (
+    <div className="page-container snap-install">
+      <div className="headers">
+        <PermissionConnectHeader
+          icon={targetSubjectMetadata.iconUrl}
+          iconName={targetSubjectMetadata.name}
+          headerTitle={t('snapInstall')}
+          headerText={null} // TODO(ritave): Add header text when snaps support description
+          siteOrigin={targetSubjectMetadata.name}
```

Comment:
The closest thing we have to an origin is the npm ID I think? The name certainly seems like the wrong thing to show here.

Reply:
This `siteOrigin` is probably not the best name for this property. It's displayed in a Chip as a name of the Snap

Reply:
On second thought, I don't think the snap should be shown in the "SiteOrigin" component at all. That design would be inconsistent, and might make it easier to mislead users with misleading snap names. I created a thread about this on Slack.